tools/buildsteps/win32/buildffmpeg.sh

```diff
@@ -12,6 +12,17 @@ FFMPEG_TARGET_OS=mingw32
 do_loaddeps $FFMPEG_VERSION_FILE
 FFMPEGDESTDIR=/xbmc/lib/win32/$LIBNAME
 
+if [ "$(pathChanged $FFMPEGDESTDIR $FFMPEG_VERSION_FILE /xbmc/project/BuildDependencies/DownloadMingwBuildEnv.bat /xbmc/tools/buildsteps/win32)" == "0" ]
```

Comment:
pathChanged only tests the first param. So you effectively only check if $FFMPEGDESTDIR changed.
You either need a loop here on in the pathChanged function